### DISCUSSION TITLE
Looping through an array and perform calculations

### DISCUSSION THREAD

**Discussion by Disura-Randunu** [schibsted/jslt#233] [Q&A] [answered]
Hello JSLT Team. Is it possible to loop through an array of objects and just perform some calculations? 
```
{
      "items":[
        {
          "number":10
        },
        {
          "number":5
        }
      ]
}
````
If we take the above payload as the input, Is it possible to get the output of the sum of "number" values so that the final output will be something like below?
```
{
   "sum_of_numbers": 15
}
```
I don't want the new array to return anything. Just want to perform an addition calculation while looping through it and store it in a new variable or something and return that value. Is this possible?

**Comment by larsga** [comment, marked as answer]
The `sum()` function will compute the sum of the values in the array passed to it, so all you need to do is make an array of the numbers, then pass it to sum.

So this will do it:

```
sum([for (.items) .number])
```

**Reply by Disura-Randunu** [reply]
It seems that I'm doing something wrong. It returns an error for the payload I'm trying to process. 

> Error:  array must contain numbers, found null

Payload:
```
{
    "name":"some_name",
    "items":[
        {
            "id": 1
            "name": "item 1"
            "fields":{
                "type": "bug",
                "timespent": 1000,
                "date": "2021-10-26"
            }
        },
        {
            "id": 1
            "name": "item 2"
            "fields":{
                "type": "bug",
                "timespent": 1000,
                "date": "2021-10-26"
            }
        },
        {
            "id": 1
            "name": "item 3"
            "fields":{
                "type": "task",
                "timespent": 500,
                "date": "2021-10-26"
            }
        }
    ]
}
```
What I want is to get the sum of "timespent" values. Also, It would be great if we can check for a condition at the same time. What I want to get is the total of "timespent" if the item's "type" value is "bug". So the final output would be 2000 since one item is not a "bug" type.

**Reply by Disura-Randunu** [reply]
Ok. I tried and got it working.
```
sum([for (.items) .fields.timespent if(.fields.type == "bug")])
```
Thank you very much for the first answer you provided.
